lib/credentials/chainable_temporary_credentials.js

```diff
@@ -0,0 +1,110 @@
+var AWS = require('../core');
+var STS = require('../../clients/sts');
+
+/**
+ * Represents temporary credentials retrieved from {AWS.STS}. Without any
+ * extra parameters, credentials will be fetched from the
+ * {AWS.STS.getSessionToken} operation. If an IAM role is provided, the
+ * {AWS.STS.assumeRole} operation will be used to fetch credentials for the
+ * role instead.
+ *
+ * To setup temporary credentials, configure a set of master credentials
+ * using the standard credentials providers (environment, EC2 instance metadata,
+ * or from the filesystem), then set the global credentials to a new
+ * temporary credentials object:
+ *
+ * ```javascript
+ * // Note that environment credentials are loaded by default,
+ * // the following line is shown for clarity:
+ * AWS.config.credentials = new AWS.EnvironmentCredentials('AWS');
+ *
+ * // Now set temporary credentials seeded from the master credentials
+ * AWS.config.credentials = new AWS.ChainableTemporaryCredentials();
```

Comment:
I've updated the docs and added support for loading an MFA token if `SerialNumber` is in the params. I also refactored the constructor interface to take an options map instead of positional arguments to provide better continuity with the other credential classes.